cmd/user-list/main_test.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/users"
+	"github.com/edulinq/autograder/internal/cmd"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+// Use the common main for all tests in this package.
+func TestMain(suite *testing.M) {
+	cmd.CMDServerTestingMain(suite)
+}
+
+func TestServerUserListBase(test *testing.T) {
+	testCases := []struct {
+		expectedNumUsers int
+		expectedExitCode int
+	}{
+		{10, 0},
+	}
+
+	serverUsersMap := db.MustGetServerUsers()
+
+	expectedEmailList := make([]string, 0, len(serverUsersMap))
+	for _, user := range serverUsersMap {
+		expectedEmailList = append(expectedEmailList, user.Email)
+	}
+
+	for i, testCase := range testCases {
+		args := []string{}
+
+		stdout, stderr, exitCode, err := cmd.RunCMDTest(test, main, args)
+		if err != nil {
+			test.Errorf("Case %d: CMD run returned an error: '%v'.", i, err)
+			continue
+		}
+
+		if len(stderr) > 0 {
+			test.Errorf("Case %d: CMD has content in stderr: '%s'.", i, stderr)
+			continue
+		}
+
+		if testCase.expectedExitCode != exitCode {
+			test.Errorf("Case %d: Unexpected exit code. Expected: '%d', Actual: '%d'.", i, testCase.expectedExitCode, exitCode)
+			continue
+		}
```

Comment:
Of course there needs to be some changes here to sync with peek's test,
but I am seeing some opportunity for common code.

This is not hard required (yet), as I am just brainstorming.

What about some functionality in internal/cmd that defines a standard test case struct,
and a function that takes the test case (and some other stuff),
does all these things things, and returns the response content (probably an error as well).

Reply:
Good idea, I can try it out to see how it feels. Should this be in a separate PR?

Reply:
No, it should be in this PR.
You are already making common tests, there's no need to do them twice.